libraries/lib-mixer/Mix.h

```diff
@@ -130,14 +137,15 @@ class MIXER_API Mixer {
  private:
 
    // Output
-   const bool       mApplyTrackGains;
+   const ApplyGain  mApplyGain;
    const bool       mHighQuality; // dithering
    const sampleFormat mFormat; // output format also influences dithering
    const bool       mInterleaved;
 
    // INPUT
    sampleFormat     mEffectiveFormat;
    bool             mNeedsDither;
+   bool             mHasMixerSpec{false};
```

Comment:
The problem is in c-tor. You can specify arguments that doesn't match each other. For example `MixerSpec` is discarded when the number of channels specified in `MixerSpec` isn't equal to number of output channels. May be if we had 2 or 3 constructors this class could have a bit more clear interface

Reply:
Yeah, ok.

So recap:
* `Discard`: gain = 1 (-> channel-wise summation)
* `MapChannels`: gain depends on level and pan
* `Mixdown`: gain depends on level and pan and number of channels

Then on top of that, the mixer spec allows to exclude some channels from some tracks from the mix.
The ctor allows to pass `Mixdown` but that won't have an effect if a valid spec is passed.

I wonder why not giving another boolean argument wouldn't have kept things a bit simpler here? Keep `bool mApplyTrackGains` but adding `mAverageChannels` ?

Reply:
Food for thoughts: maybe it would make the internals of the mixer easier if the gains were part of the mixer spec, that is, it wouldn't be a matrix of booleans anymore but of doubles. It wouldn't be an optional argument anymore, and some external, testable helper could construct it.

Reply:
> I wonder why not giving another boolean argument wouldn't have kept things a bit simpler here? Keep bool mApplyTrackGains but adding mAverageChannels ?

That would give 8 possible combinations of input parameters (2 states for mixer spec, 2 for `mApplyTrackGains` and 2 for `mAverageChannels`), instead of 6 when we use 3-state enum.